sqlrunner/main.py

```diff
@@ -35,7 +37,13 @@ def main(args):
 
 
 def generate_column_headers(dummy_data_file, sql_query):
-    raise NotImplementedError
+    tsql_parser = TSQL()
+    parsed = [q for q in tsql_parser.parse(sql_query) if q]
```

Comment:
I think we're filtering out `None`s here, aren't we? In which case

```py
    parsed = [q for q in tsql_parser.parse(sql_query) if q is not None]
```

would be clearer.

Reply:
Yes, it seems to produce additional elements which are `None` - happy to change if you think clearer

Reply:
Do you know why it produces these elements? Can the parser be configured not to do so?

Reply:
AFAICT it happens in the test case because there is a `;` preceding the `WITH` of the CTE (which is a requirement with mssql). The parser sees this as a blank statement terminated by the semicolon and then the next one.

There may be legitimate cases where there are multiple statements in a batch - e.g. putting some data into temp tables and then selecting from them. In this case, we actually want the last not-null parsed `Expression` not the first so this current implementation is wrong. 

We _could_ rely on taking the last statement in the batch as the one for which we wish to generate headers but it's not inconceivable that someone might add a comment or a stray blank line or something at the end which might generate an empty expression as the last one in the parsed list. I think it's safer to filter the empty expressions.

Reply:
I didn't know that about `;` and `WITH`. I've certainly written the latter without the former, so perhaps pymssql is adding them. Either way, it's good to filter out empty statements. Perhaps you could add a comment?

Let's cross the multiple statements bridge when we come to it 🙂

Reply:
I'd already added a test case for multiple statements before I'd read this comment so I'm leaving that in for now. I've added a couple of comments to explain this filtering.